File: 백준/Silver/1302. 베스트셀러/베스트셀러.py
N = int(input())
book_dict = dict()
for _ in range(N):
    book = input()
    if book in book_dict:
        book_dict[book] += 1
    else:
        book_dict[book] = 1

answer = []
max_cnt = 0
for k, v in book_dict.items():
    if v > max_cnt:
        max_cnt = v
        answer = [k]
    elif v == max_cnt:
        answer.append(k)

answer.sort()
print(answer[0])

File: 백준/Silver/1302. 베스트셀러/베스트셀러.java
import java.io.*;
import java.util.HashMap;
import java.util.ArrayList;
import java.util.Map;
import java.util.Map.Entry;
import java.util.Collections;
public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        Map<String, Integer> bookSale = new HashMap<>();
        int N = Integer.parseInt(br.readLine());
        for (int i = 0; i < N; i++) {
            String book = br.readLine();
            if (bookSale.containsKey(book)) {
                bookSale.put(book, bookSale.get(book) + 1);
            } else {
                bookSale.put(book, 1);
            }
        }
        int max_cnt = 0;
        ArrayList<String> books = new ArrayList<>();
        for (Entry<String, Integer> elem : bookSale.entrySet()) {
            String b = elem.getKey();
            int cnt = elem.getValue();
            if (cnt > max_cnt) {
                max_cnt = cnt;
                books = new ArrayList<>();
                books.add(b);
            } else if (cnt == max_cnt) {
                books.add(b);
            }
        }
        Collections.sort(books);
        System.out.println(books.get(0));
    }
}
